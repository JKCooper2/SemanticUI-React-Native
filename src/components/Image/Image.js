// @flow
import React, { PureComponent } from 'react';
import { Image } from 'react-native';

import _ from 'lodash';

import type { ImagePropsType } from '#/components/Image';

let SIZES = {
    mini: 20,
    tiny: 50,
    small: 100,
    medium: 180,
    large: 300,
    big: 450,
    huge: 600,
    massive: 800
};

class Icon extends PureComponent<ImagePropsType> {
    static defaultProps = {
        size: 'medium',
    };

    render() {
        const { src, size } = this.props;

        let width = _.get(SIZES, size, SIZES.medium);

        return (
            <Image
                source={{ uri: src }}
                style={{ width, height: width }}
                resizeMode="contain"
            />
        );
    }
}

export default Icon;