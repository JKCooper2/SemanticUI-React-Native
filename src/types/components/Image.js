// @flow
type SizeType = 'mini' | 'tiny' | 'small' | 'medium' | 'large' | 'big' | 'huge' | 'massive';

export type ImagePropsType = {
    src: string,
    size: SizeType
};
