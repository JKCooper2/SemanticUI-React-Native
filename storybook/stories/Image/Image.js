// @flow
import React, { Fragment } from 'react';
import { View, Text, ScrollView } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import Image from '~/components/Image';

const imageStories = storiesOf('Image', module);

imageStories.add('Standard', () => (
    <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="small"/>
));

imageStories.add('Sizes', () => (
    <ScrollView>
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="mini" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="tiny" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="small" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="medium" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="large" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="big" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="huge" />
        <Image src="https://react.semantic-ui.com/images/wireframe/image.png" size="massive" />
    </ScrollView>
));

export default imageStories;
