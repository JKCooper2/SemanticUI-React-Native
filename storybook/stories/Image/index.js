// @flow
import Image from './Image';

export {
    Image
};