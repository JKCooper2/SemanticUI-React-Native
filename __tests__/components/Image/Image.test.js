// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Image from '~/components/Image';

test('renders correctly', () => {
    const image = shallow(<Image src="https://react.semantic-ui.com/logo.png" />);
    expect(image).toMatchSnapshot();
});
