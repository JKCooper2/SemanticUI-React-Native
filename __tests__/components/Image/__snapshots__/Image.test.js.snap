// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  resizeMode="contain"
  source={
    Object {
      "uri": "https://react.semantic-ui.com/logo.png",
    }
  }
  style={
    Object {
      "height": 180,
      "width": 180,
    }
  }
/>
`;
